src/codeu/chat/server/Model.java

```diff
@@ -67,6 +69,27 @@ public int compare(Time a, Time b) {
   private final Store<Time, Message> messageByTime = new Store<>(TIME_COMPARE);
   private final Store<String, Message> messageByText = new Store<>(STRING_COMPARE);
 
+  private final Store<User, HashSet<Uuid>> userInterestByUser = new Store<>(null);
+  private final Store<User, HashSet<Uuid>> convoInterestByUser = new Store<>(null);
```

Comment:
Because `UserInterest` already has the two hashets:

https://github.com/lianasyrkett/CodeU-Summer-2017/pull/4/files#diff-3392905fb69f7197da51b37ce9cc11cf

The value you store should be a UserInterest object, which will have the hashets for users and convos that the user is interested int.